File: samplejava/src/main/java/com/example/chattutorial/MainActivity.java
package com.example.chattutorial;

import static java.util.Collections.singletonList;

import android.os.Bundle;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;
import androidx.lifecycle.ViewModelProvider;

import com.example.chattutorial.databinding.ActivityMainBinding;

import org.jetbrains.annotations.Nullable;

import io.getstream.chat.android.client.ChatClient;
import io.getstream.chat.android.client.logger.ChatLogLevel;
import io.getstream.chat.android.models.FilterObject;
import io.getstream.chat.android.models.Filters;
import io.getstream.chat.android.models.User;
import io.getstream.chat.android.offline.plugin.factory.StreamOfflinePluginFactory;
import io.getstream.chat.android.state.plugin.config.StatePluginConfig;
import io.getstream.chat.android.state.plugin.factory.StreamStatePluginFactory;
import io.getstream.chat.android.ui.viewmodel.channels.ChannelListViewModel;
import io.getstream.chat.android.ui.viewmodel.channels.ChannelListViewModelBinding;
import io.getstream.chat.android.ui.viewmodel.channels.ChannelListViewModelFactory;

public final class MainActivity extends AppCompatActivity {

    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        // Step 0 - inflate binding
        ActivityMainBinding binding = ActivityMainBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());

        // Step 1 - Set up the OfflinePlugin for offline storage
        StreamOfflinePluginFactory streamOfflinePluginFactory = new StreamOfflinePluginFactory(
                getApplicationContext()
        );
        StreamStatePluginFactory streamStatePluginFactory = new StreamStatePluginFactory(
                new StatePluginConfig(true, true), this
        );

        // Step 2 - Set up the client for API calls with the plugin for offline storage
        ChatClient client = new ChatClient.Builder("hz7v96dxqebs", getApplicationContext())
                .withPlugins(streamOfflinePluginFactory, streamStatePluginFactory)
                .logLevel(ChatLogLevel.ALL) // Set to NOTHING in prod
                .build();

        // Step 3 - Authenticate and connect the user
        // Step 3 - Authenticate and connect the user
        User user = new User.Builder()
                .withId("ach123")
                .withName("student121")
                .withImage("https://bit.ly/2TIt8NR")
                .build();

        client.connectUser(
                user,
                "zwx2hmke7knubvfhpx98fzt9qp3x7ttjhu5uvgj4gyevkfgf95uygcexybyp2rf6"
        ).enqueue(result -> {
            if (result.isSuccess()) {
                // User is successfully connected, proceed with creating ChannelListViewModelFactory

                // Step 4 - Set the channel list filter and order
                FilterObject filter = Filters.and(
                        Filters.eq("type", "messaging"),
                        Filters.in("members", singletonList(user.getId()))
                );

                ViewModelProvider.Factory factory = new ChannelListViewModelFactory.Builder()
                        .filter(filter)
                        .sort(ChannelListViewModel.DEFAULT_SORT)
                        .build();

                ChannelListViewModel channelsViewModel =
                        new ViewModelProvider(this, factory).get(ChannelListViewModel.class);

                // Step 5 - Connect the ChannelListViewModel to the ChannelListView
                ChannelListViewModelBinding.bind(channelsViewModel, binding.channelListView, this);
                binding.channelListView.setChannelItemClickListener(
                        channel -> startActivity(ChannelActivity4.newIntent(this, channel))
                );
            } else {
                // Handle the case where connecting the user failed
                Toast.makeText(this, "Failed to connect user!", Toast.LENGTH_SHORT).show();
            }
        });
    }
}

File: samplekotlin/src/main/java/com/example/chattutorial/MainActivity.kt
package com.example.chattutorial
import android.os.Bundle
import android.widget.Toast
import androidx.activity.viewModels
import androidx.appcompat.app.AppCompatActivity
import com.example.chattutorial.databinding.ActivityMainBinding
import com.google.android.material.floatingactionbutton.FloatingActionButton
import io.getstream.chat.android.client.ChatClient
import io.getstream.chat.android.client.logger.ChatLogLevel
import io.getstream.chat.android.models.Filters
import io.getstream.chat.android.models.User
import io.getstream.chat.android.offline.plugin.factory.StreamOfflinePluginFactory
import io.getstream.chat.android.state.plugin.config.StatePluginConfig
import io.getstream.chat.android.state.plugin.factory.StreamStatePluginFactory
import io.getstream.chat.android.ui.viewmodel.channels.ChannelListViewModel
import io.getstream.chat.android.ui.viewmodel.channels.ChannelListViewModelFactory
import io.getstream.chat.android.ui.viewmodel.channels.bindView


class MainActivity : AppCompatActivity() {

    private lateinit var binding: ActivityMainBinding

    override fun onCreate(savedInstanceState: Bundle?) {
//        AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_FOLLOW_SYSTEM)
//        toggleDarkMode()
        super.onCreate(savedInstanceState)

        // Step 0 - inflate binding
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)

        // Step 1 - Set up the OfflinePlugin for offline storage
        val offlinePluginFactory = StreamOfflinePluginFactory(appContext = this)
        val statePluginFactory = StreamStatePluginFactory(
            config = StatePluginConfig(
                backgroundSyncEnabled = true,
                userPresence = true,
            ),
            appContext = this,
        )

        // Step 2 - Set up the client for API calls with the plugin for offline storage
        val client = ChatClient.Builder("f7ypjdv2usn9", applicationContext)
            .withPlugins(offlinePluginFactory, statePluginFactory)
            .logLevel(ChatLogLevel.ALL) // Set to NOTHING in prod
            .build()
        Toast.makeText(this, client.toString(), Toast.LENGTH_SHORT).show()
        // Step 3 - Authenticate and connect the user
        val user = User(
            id = "student121",
            name = "student121",
            image = ""
        )
//        client.searchMessages()
        val t=client.devToken(userId = "student121")
//        print(user)
        client.connectUser(
            user = user,
            token = t
        ).enqueue {
            if (it.isSuccess) {
                // Step 4 - Set the channel list filter and order
                // This can be read as requiring only channels whose "type" is "messaging" AND
                // whose "members" include our "user.id"
                val filter = Filters.and(
                    Filters.eq("type", "messaging"),
                    Filters.`in`("members", listOf(user.id))
                )
                val viewModelFactory =
                    ChannelListViewModelFactory(filter, ChannelListViewModel.DEFAULT_SORT)
                val viewModel: ChannelListViewModel by viewModels { viewModelFactory }

                // Step 5 - Connect the ChannelListViewModel to the ChannelListView, loose
                //          coupling makes it easy to customize
                viewModel.bindView(binding.channelListView, this)
                binding.channelListView.setChannelItemClickListener { channel ->
                    startActivity(ChannelActivity.newIntent(this, channel))
                }
            } else {
                Toast.makeText(this, "something went wrong!", Toast.LENGTH_SHORT).show()
            }


            val fab = findViewById<FloatingActionButton>(R.id.floatingActionButton)
            fab.setOnClickListener { view ->
                val channelClient =
                    client.channel(channelType = "messaging", channelId = "general2")

                // Watch the channel
                channelClient.watch().enqueue { result ->
                    if (result.isSuccess) {
                        // Channel successfully watched, now add members
                        channelClient.addMembers(listOf("ach123","shru")).enqueue { addMembersResult ->
                            if (addMembersResult.isSuccess) {
                                // Members added successfully
                            } else {
                                // Handle addMembersResult.error()
                            }
                        }
                    } else {
                        // Handle result.error()
                    }
                }
            }
        }













        }
    }

